app/models/rich_text_reaction.rb

```diff
@@ -0,0 +1,54 @@
+# == Schema Information
+#
+# Table name: rich_text_reactions
+#
+#  id            :bigint           not null, primary key
+#  emoji_caption :string           not null
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  rich_text_id  :bigint           not null
+#  user_id       :bigint           not null
+#
+# Indexes
+#
+#  index_rich_text_reactions_on_rich_text_id  (rich_text_id)
+#  index_rich_text_reactions_on_user_id       (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (rich_text_id => action_text_rich_texts.id)
+#  fk_rails_...  (user_id => users.id)
+#
+class RichTextReaction < ApplicationRecord
+  # Permissible set of emojis for standup updates and comments.
+  EMOJI_DICT = {
+    'thumbs_up' => '👍',
+    'thinking' => '🤔',
+    'hooray' => '🎉',
+    'shrug' => '🤷',
+    'thumbs_down' => '👎',
+    'eyes' => '👀'
+  }.freeze
+
+  belongs_to :user
+  belongs_to :rich_text, class_name: 'ActionText::RichText'
+
+  validates :emoji_caption,
+    inclusion: {
+      in: EMOJI_DICT.keys,
+      message: 'must be present in permissible set'
+    }
+
+  # Return the emoji icon for the caption.
+  # e.g. '👍' for 'thumbs_up'
+  def emoji
+    EMOJI_DICT[emoji_caption]
+  end
+
+  class << self
```

Comment:
Personally I find the syntax `def self.emoji_captions` easier to understand.

Reply:
Here are two reasons why I’ve learned to prefer this style:
1. I like how `class << self` allows me to physically organize my class methods, especially with private/protected methods.
2. I like looking at class methods like I would another class definition since class methods are merely instance methods of the singleton class object for the class we’re working inside.

Having said that, I prefer consistency among team members, and would be glad to change this.
Just let me know. 👍

Reply:
All good as is then. Thanks for the clarification!